src/Microsoft.AspNet.OData.Shared/ODataAPIHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ODataAPIHandler.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Handler Class to handle users methods for create, delete and update.
+    /// This is the handler for data modification where there is a CLR type.
+    /// </summary>
+    internal abstract class ODataAPIHandler<TStructuralType>: IODataAPIHandler where TStructuralType : class
```

Comment:
It seems weird that we are constraining the generic type parameter to just `class`. Why is that restriction necessary?

Reply:
It is used to handle delta request and it should be only for class types